### PR TITLE
docs: add tcp vs udp part

### PR DESCRIPTION
**Pull request by mateuszpierzchala-splunk**

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -41,6 +41,22 @@ _will_ be data loss (think CD-quality (lossless) vs. MP3).  Syslog data collecti
...
-Paradoxically, UDP for syslog actually ends up being a better choice for resiliency for syslog.  For an excellent discussion on this topic
-(as well as the "myth" of load balancers for HA),
-see [Performant AND Reliable Syslog: UDP is best](https://www.rfaircloth.com/2020/05/21/performant-and-reliable-syslog-udp-is-best/).
+For running syslog UDP is recommended over TCP.
+
+The syslogd daemon was originally configured to use UDP for log forwarding to reduce overhead. 
+While UDP is an unreliable protocol, it's streaming method does not require the overhead of establishing a network session. 
+This protocol also reduces network load as the network stream with no required receipt verification or window adjustment.
+While TCP could seem a better choice because it uses ACKS and there should not be data loss, there are some cases when it's possible:
+* the TCP session is closed events published while the system is creating a new session will be lost. (Closed Window Case)
```
Comment (rjha-splunk) [inline docs/architecture.md:50]: The TCP session

File: docs/architecture.md [hunk 2]
```diff
@@ -41,6 +41,22 @@ _will_ be data loss (think CD-quality (lossless) vs. MP3).  Syslog data collecti
...
+* the remote side is busy and can not ack fast enough events are lost due to local buffer full
```
Comment (rjha-splunk) [inline docs/architecture.md:51]: The

File: docs/architecture.md [hunk 3]
```diff
@@ -41,6 +41,22 @@ _will_ be data loss (think CD-quality (lossless) vs. MP3).  Syslog data collecti
...
+* a single ack is lost by the network and the client closes the connection. (local and remote buffer lost)
```
Comment (rjha-splunk) [inline docs/architecture.md:52]: A

File: docs/architecture.md [hunk 4]
```diff
@@ -41,6 +41,22 @@ _will_ be data loss (think CD-quality (lossless) vs. MP3).  Syslog data collecti
...
+* the remote server restarts for any reason (local buffer lost)
+* the remote server restarts without closing the connection (local buffer plus timeout time lost)
```
Comment (rjha-splunk) [inline docs/architecture.md:54]: The

File: docs/architecture.md [hunk 5]
```diff
@@ -41,6 +41,22 @@ _will_ be data loss (think CD-quality (lossless) vs. MP3).  Syslog data collecti
...
+For running syslog UDP is recommended over TCP.
+
+The syslogd daemon was originally configured to use UDP for log forwarding to reduce overhead. 
+While UDP is an unreliable protocol, it's streaming method does not require the overhead of establishing a network session. 
+This protocol also reduces network load as the network stream with no required receipt verification or window adjustment.
+While TCP could seem a better choice because it uses ACKS and there should not be data loss, there are some cases when it's possible:
+* the TCP session is closed events published while the system is creating a new session will be lost. (Closed Window Case)
+* the remote side is busy and can not ack fast enough events are lost due to local buffer full
+* a single ack is lost by the network and the client closes the connection. (local and remote buffer lost)
+* the remote server restarts for any reason (local buffer lost)
```
Comment (rjha-splunk) [inline docs/architecture.md:53]: The

**Comment by mateuszpierzchala-splunk** [comment]
closes #1045

**Pull request merged**